### PR TITLE
Add django-dev-dashboard as an app

### LINKED ISSUE #309: Move dev dashboard code into this repo
There just was this PR to update the dev dashboard to the new styles: https://github.com/django/django-dev-dashboard/pull/15

And I think similar to the Trac issue we'll have a hard time keeping them in sync separately. Plus, Jacob has indicated that he wants to do less not more Django work, so I guess we better move this over. What do you think? We can handle the same stuff with our new multiple host stuff based on django-hosts.

### PR DESCRIPTION
**Pull request by relekang**
It uses the code from the django/django-dev-dashboard with some design changes and new migrations. It is added under the host dashboard.djangoproject.com.

There is still some work needed to be done, but I opened this in hope of getting som feedback.

screenshot 2015-01-16 10 49 46

Closes #309 

**Todo:**
- [x] Add tests for the python code
- [x] Install jquery-flot with bower
- [ ] Update ansible playbook to create update_metrics cronjob

### CODE REVIEW

**Comment by jezdez** [comment]
Nice work!

One thing I noticed is that it doesn't use the bower setup for the jquery plugin. Any specific reason?

File: dashboard/views.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    try:
+        daysback = int(request.GET['days'])
+    except (TypeError, KeyError, ValueError):
+        daysback = 30
+    d = datetime.datetime.now() - datetime.timedelta(days=daysback)
+
+    doc = model_to_dict(metric)
+    doc['data'] = metric.gather_data(since=d)
+
+    return http.HttpResponse(
```
Comment (jezdez) [inline dashboard/views.py]: This could use the new JSONResponse class

**Comment by relekang** [comment]
> One thing I noticed is that it doesn't use the bower setup for the jquery plugin. Any specific reason?

Only because I haven't found a good way to build it yet, bower clones the gitrepo without minified versions or a concatenated version. It could probably be done in the Makefile?

**Comment by jezdez** [comment]
@relekang Nicely done! The Makefile approach is good enough but we're serving the files with gzip so there is really not much sense in minifying the files (or at least very little). I don't mind having the make task around, since we definitely have inconsistency with some of the other dependencies, so maybe we can revisit this whole minifying business at a later in a different branch/PR.

File: dashboard/urls.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+from django.conf.urls import patterns, include, url
+from django.contrib import admin
+
+urlpatterns = patterns(
+    '',
+    url('^$',                       'dashboard.views.index',            name="dashboard-index"),
```
Comment (jezdez) [inline dashboard/urls.py]: Please don't block indent

File: dashboard/urls.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    url('^metric/([\w-]+)/$',       'dashboard.views.metric_detail',    name="metric-detail"),
+    url('^metric/([\w-]+).json$',   'dashboard.views.metric_json',      name="metric-json"),
+    url(r'^admin/', include(admin.site.urls)),
```
Comment (jezdez) [inline dashboard/urls.py]: The admin URL isn't needed

File: dashboard/urls.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+from django.conf.urls import patterns, include, url
+from django.contrib import admin
+
+urlpatterns = patterns(
```
Comment (jezdez) [inline dashboard/urls.py]: Could you make urlpatterns just a list? The patterns function is deprecated

File: dashboard/views.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+from django.shortcuts import render
+from django.forms.models import model_to_dict
+from django.views.decorators.cache import cache_page
+
+from .models import Metric
+
+TEN_MINUTES = 60 * 10
+
+
+@cache_page(TEN_MINUTES)
```
Comment (jezdez) [inline dashboard/views.py]: We're using the full page cache middleware right now, so I'm wondering how applying the cache_page decorator here will affect the caching of those views? I think we can safely remove them

File: dashboard/views.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    return JsonResponse(doc)
+
+
+def _find_metric_or_404(slug):
+    for MC in Metric.__subclasses__():
+        try:
+            return MC.objects.get(slug=slug)
+        except MC.DoesNotExist:
+            continue
+    raise Http404()
```
Comment (jezdez) [inline dashboard/views.py]: Mind passing in a reason why this 404s? Like `raise Http404('Could not find metric with slug %s' % slug)` or something?

File: djangoproject/templates/dashboard/index.html [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        <a href="{{ report.metric.link }}">{{ report.latest.measurement }}{% if report.metric.unit == "%" %}%{% endif %}</a>
+        <div class="timestamp">&nbsp;</div>
+      </div>
+      {% if report.metric.show_sparkline %}
+        <div class="sparkline" id="spark{{ forloop.counter0 }}" data-metric="{{ report.metric.slug }}"></div>
+      {% endif %}
+    </div>
+  {% endfor %}
+  <p class="updated">
+    Updated {{ data.0.latest.timestamp|timesince }} ago.
```
Comment (jezdez) [inline djangoproject/templates/dashboard/index.html]: Needs to be marked for translation

File: djangoproject/templates/base_dashboard.html [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{% extends "base.html" %}
```
Comment (jezdez) [inline djangoproject/templates/base_dashboard.html]: Strings need to be marked for translation

**Comment by jezdez** [comment]
I'm gonna take care of setting up the cronjob for running the update_metric command hourly

**Comment by relekang** [comment]
Thanks for the input

**Comment by relekang** [comment]
@jezdez I added some tests

**Comment by jezdez** [comment]
Sorry that this has gotten a bit on the backburner, we've been working on a fundraising page in the past week but launched today. Gonna try getting this properly reviewed again and pushed as soon as possible. Thank you for your patience, @relekang!

**Comment by relekang** [comment]
I understand, thank you for letting me now :)

I'll try to keep this up to date with master.

**Comment by jezdez** [comment]
I've made some minor updates (e.g. use requests-mock instead of rmoq) and merged master in #363.

**Pull request closed without merging**